src/vs/editor/common/services/modelService.ts

```diff
@@ -710,6 +710,14 @@ class SemanticColoringFeature extends Disposable {
 		}));
 		this._register(themeService.onDidColorThemeChange(handleSettingOrThemeChange));
 	}
+
+	override dispose(): void {
+		// Dispose all watchers
+		for (const watcher of Object.values(this._watchers)) {
+			watcher.dispose();
+		}
```

Comment:
just fyi that the `dispose`-utility support iterables, eg `dispose(Object.values(this._watchers))` would work

Reply:
:+1: Good point. This regex finds around 50 usages that might be replaced `for.*\{\n\s*.*\.dispose\(\);\n\s*\}`